proxy/httpproxy/reverse.go

```diff
@@ -112,9 +112,10 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	closeNotifier, ok := rw.(http.CloseNotifier)
```

Comment:
The response writer is closed before the goroutine is scheduled. This nils out some internal state of the response writer so when the goroutine finally runs CloseNotify() it panics.